### PR TITLE
Zadania z adaptera

### PR DESCRIPTION
**Pull request by BartoszKowalczyk98**
indeks: 215772

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawControllerAdapter.java [hunk 1]
```diff
@@ -5,13 +5,15 @@
 import edu.kis.legacy.drawer.shape.LineFactory;
 import edu.kis.powp.jobs2d.Job2dDriver;
 
+import static edu.kis.powp.jobs2d.features.DrawerFeature.getDrawerController;
+
 /**
  * driver adapter to drawer with several bugs.
  */
-public class MyAdapter extends DrawPanelController implements Job2dDriver {
+public class DrawControllerAdapter extends DrawPanelController implements Job2dDriver {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawControllerAdapter.java]: The name is better but it does not indicate the purpose of the adapter or what was actually adapted.

Inheritance is not used (no super drawLine calls).

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawControllerAdapter.java [hunk 2]
```diff
@@ -5,13 +5,15 @@
 import edu.kis.legacy.drawer.shape.LineFactory;
 import edu.kis.powp.jobs2d.Job2dDriver;
 
+import static edu.kis.powp.jobs2d.features.DrawerFeature.getDrawerController;
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawControllerAdapter.java]: Direct coupling with DrawerFeature class is unnecessary.

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package edu.kis.powp.jobs2d.drivers.adapter;
+
+import edu.kis.legacy.drawer.panel.DrawPanelController;
+import edu.kis.legacy.drawer.shape.ILine;
+import edu.kis.legacy.drawer.shape.LineFactory;
+import edu.kis.powp.jobs2d.Job2dDriver;
+
+import static edu.kis.powp.jobs2d.features.DrawerFeature.getDrawerController;
+
+public class LineDrawerAdapter extends DrawPanelController implements Job2dDriver{
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java]: Similar issues.

File: src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java [hunk 4]
```diff
@@ -16,6 +16,11 @@ public SelectTestFigureOptionListener(DriverManager driverManager) {
 
 	@Override
 	public void actionPerformed(ActionEvent e) {
-		FiguresJoe.figureScript1(driverManager.getCurrentDriver());
+
+		if(e.paramString().contains("Figure Joe 1"))
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java]: Menu action should not depend on its caption. Enum or separate listeners would be better. At least you should consider constants.

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+    @Override
+    public void setPosition(int x, int y) {
+        this.startX = x;
+        this.startY = y;
+    }
+
+    @Override
+    public void operateTo(int x, int y) {
+        ILine line = LineFactory.getSpecialLine();
```
Comment (rsusik) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java:26]: Unfortunately, this is not a generic solution (e.g. new line type requires new class, unnecessary direct coupling).

**Review by rsusik — COMMENTED** [review]
Specific remarks are inlined in the code.

**Pull request closed without merging**